packages/plan-form/src/Form/componentsUtils/status.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Radio, Popconfirm } from 'antd';
+import { PlanStatus, planStatusDisplay, status } from '@opensrp/plan-form-core';
+import { FormInstance } from 'antd/lib/form';
+import {
+  NO,
+  SETTING_STATUS_TO_ACTIVE,
+  SETTING_STATUS_TO_COMPLETE,
+  SETTING_STATUS_TO_DRAFT,
+  SETTING_STATUS_TO_RETIRED,
+  YES,
+} from '../../lang';
+
+interface PlanStatusRendererProps {
+  disabledFields: string[];
+  disAllowedStatusChoices: string[];
+  setFieldsValue: FormInstance['setFieldsValue'];
+}
+
+const popUpConfirmationMessageLookup = {
+  [PlanStatus.DRAFT]: SETTING_STATUS_TO_DRAFT,
+  [PlanStatus.ACTIVE]: SETTING_STATUS_TO_ACTIVE,
+  [PlanStatus.COMPLETE]: SETTING_STATUS_TO_COMPLETE,
+  [PlanStatus.RETIRED]: SETTING_STATUS_TO_RETIRED,
+};
+
+/** renders the status fields on the planForm
+ * Wraps the radio buttons in a pop up confirmation box, this means
+ * we need some custom logic upstream to change how antd form updates
+ * values, so that when user does not confirm in the popup, we retain
+ * the previous value.
+ *
+ * @param props - this components props
+ */
+export const PlanStatusRenderer = (props: PlanStatusRendererProps) => {
+  const { disabledFields, disAllowedStatusChoices, setFieldsValue, ...rest } = props;
+  return (
+    <Radio.Group className="plan-form-status" {...rest} disabled={disabledFields.includes(status)}>
+      {Object.entries(PlanStatus)
+        .filter((e) => !disAllowedStatusChoices.includes(e[1]))
+        .map((e) => {
+          const titleMessage = () => {
+            return <p className="popup-width">{popUpConfirmationMessageLookup[e[1]]}</p>;
+          };
+
+          return (
+            <Popconfirm
```

Comment:
Do we need to support something like [this](https://github.com/onaio/reveal-frontend/pull/1493)?

Reply:
just following what is requested in the mockups, though I think its nice to know that this is also an option.
cc @cafootitt https://github.com/onaio/reveal-frontend/pull/1493#issuecomment-776511430

Reply:
@p-netm that's fine but what are the plans/issues to keep this in sync across the projects?